lib/MusicBrainz/Server/Edit/Generic/Merge.pm

```diff
@@ -65,16 +65,36 @@ sub build_display_data
     my ($self, $loaded) = @_;
     my $model = $self->_merge_model;
 
-    my $new_entity = (
+    # For release merges, we need to load additional data
+    # before turning the releases to JSON
+    if ($model eq 'Release') {
+        $self->c->model('Label')->load(
+            grep { $_->label_id && !defined($_->label) }
+            map { $_->all_labels }
+            values %{ $loaded->{Release} }
```

Comment:
Can you store this in a `my @releases = values %{ $loaded->{Release} };` first? (Since it's used four times.)